src/main/java/it/pagopa/transactions/utils/NodoOperations.java

```diff
@@ -122,29 +132,38 @@ private Mono<PaymentRequestInfo> nodoActivationForNM3PaymentRequest(
-                                                ctTransferPSPV2.getMetadata().getMapEntry().parallelStream()
-                                                        .allMatch(
-                                                                ctMapEntry -> ctMapEntry.getKey()
-                                                                        .equals("IBANAPPOGGIO")
-                                                        )
-                                                &&
-                                                (isIbanCCP(ctTransferPSPV2.getIBAN())
-                                                        ||
-                                                        ctTransferPSPV2.getMetadata().getMapEntry()
-                                                                .parallelStream()
-                                                                .anyMatch(
-                                                                        ctMapEntry -> ctMapEntry
-                                                                                .getKey()
-                                                                                .equals("IBANAPPOGGIO")
-                                                                                && isIbanCCP(ctMapEntry.getValue())
-                                                                ))
-                                )
+                                isAllCCP(response, lightAllCCPCheck)
                         )
                 );
     }
 
+    private boolean isAllCCP(
+                             ActivatePaymentNoticeV2Response response,
+                             boolean lightAllCCPCheck
+    ) {
+        return lightAllCCPCheck
+                ? response.getTransferList().getTransfer().parallelStream().allMatch(t -> isIbanCCP(t.getIBAN()))
+                : response.getTransferList().getTransfer().parallelStream().allMatch(
+                        ctTransferPSPV2 -> ctTransferPSPV2.getMetadata() != null &&
+                                ctTransferPSPV2.getMetadata().getMapEntry().parallelStream()
+                                        .allMatch(
+                                                ctMapEntry -> ctMapEntry.getKey()
+                                                        .equals("IBANAPPOGGIO")
+                                        )
+                                &&
+                                (isIbanCCP(ctTransferPSPV2.getIBAN())
+                                        ||
+                                        ctTransferPSPV2.getMetadata().getMapEntry()
+                                                .parallelStream()
+                                                .anyMatch(
+                                                        ctMapEntry -> ctMapEntry
+                                                                .getKey()
+                                                                .equals("IBANAPPOGGIO")
```

Comment:
It is not blocking but we could create a constant for "IBANAPPOGGIO". What do you think ? @ciuffagianluca

Reply:
It's a good idea